sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,171 @@
+# Authors: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#          Raghav RV <rvraghav93@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import print_function
+
+from nose.tools import assert_true
+import inspect
+import warnings
+import importlib
+
+from pkgutil import walk_packages
+from inspect import getsource
+
+import sklearn
+from sklearn.base import signature
+from sklearn.utils.testing import SkipTest
+from sklearn.utils.testing import check_parameters_match
+from sklearn.utils.testing import get_func_name
+from sklearn.utils.testing import ignore_warnings
+
+_DOC_SPECIAL_MEMBERS = ('__contains__', '__getitem__', '__iter__', '__len__',
+                        '__call__', '__add__', '__sub__', '__mul__', '__div__',
+                        '__neg__', '__hash__')
+
+
+# TODO Uncomment all modules and fix doc inconsistencies everywhere
+PUBLIC_MODULES = [
+    # the list of modules users need to access for all functionality
+    # 'sklearn',
+    'sklearn.base',
+    'sklearn.calibration',
+    'sklearn.cluster',
+    'sklearn.covariance',
+    # 'sklearn.cross_decomposition',
+    'sklearn.datasets',
+    'sklearn.decomposition',
+    # 'sklearn.ensemble',
+    'sklearn.feature_extraction',
+    # 'sklearn.feature_selection',
+    'sklearn.gaussian_process',
+    'sklearn.isotonic',
+    'sklearn.linear_model',
+    'sklearn.manifold',
+    'sklearn.multiclass',
+    'sklearn.metrics',
+    'sklearn.naive_bayes',
+    'sklearn.mixture',
+    # 'sklearn.model_selection',
+    'sklearn.neighbors',
+    'sklearn.neural_network',
+    'sklearn.preprocessing',
+    'sklearn.pipeline',
+    'sklearn.semi_supervised',
+    'sklearn.tree',
+    # 'sklearn.utils',
+]
+
+# functions to ignore args / docstring of
+_DOCSTRING_IGNORES = [
+    'sklearn.utils.deprecation.load_mlcomp',
+    'sklearn.pipeline.make_pipeline',
+    'sklearn.pipeline.make_union',
+    'sklearn.utils.extmath.safe_sparse_dot',
+    # Deprecated classes and functions
+    'RandomizedPCA',
+    'GaussianProcess',
+    'VBGMM',
+    'DPGMM',
+    'GMM',
+    'log_multivariate_normal_density',
+    'sample_gaussian',
+]
+
+_TAB_IGNORES = [
+]
+
+# Methods to test for, in any class
+_METHODS_IGNORE_NONE_Y = [
+        'fit',
+        'score',
+        'fit_predict',
+        'fit_transform',
+        'partial_fit',
+        'predict'
+]
+
+
+def test_docstring_parameters():
+    """Test module docstring formatting."""
+    try:
+        import numpydoc  # noqa
+    except ImportError:
+        raise SkipTest(
+            "numpydoc is required to test the docstrings")
+
+    from numpydoc import docscrape
+
+    incorrect = []
+    for name in PUBLIC_MODULES:
+        with warnings.catch_warnings(record=True):  # traits warnings
+            module = __import__(name, globals())
+        for submod in name.split('.')[1:]:
```

Comment:
why not https://docs.python.org/3/library/importlib.html#importlib.import_module